airflow/decorators/task_group.py

```diff
@@ -124,14 +124,14 @@ def partial(self, **kwargs: Any) -> _TaskGroupFactory[FParams, FReturn]:
         # TODO: fixme when mypy gets compatible with new attrs
         return attr.evolve(self, partial_kwargs=kwargs)  # type: ignore[arg-type]
 
-    def expand(self, **kwargs: OperatorExpandArgument) -> DAGNode:
+    def expand(self, concurrency_limit: int = None, **kwargs: OperatorExpandArgument) -> DAGNode:
```

Comment:
Got it, I'll move the parameter to partial().

I previously thought about this.
At that time, my understand is the partial() pass the common parameter to mapped operators, and expand() pass the parameter for expanding.
So I put the expanding concurrency limitation to expand()...

Reply:
@uranusjr I got a little confused when I move parameter to `partial()`.

Below is the partial() definition in decorators/task_group.py:
```python
    def partial(self, **kwargs: Any) -> _TaskGroupFactory[FParams, FReturn]:
        self._validate_arg_names("partial", kwargs)
        prevent_duplicates(self.partial_kwargs, kwargs, fail_reason="duplicate partial")
        kwargs.update(self.partial_kwargs)
        # TODO: fixme when mypy gets compatible with new attrs
        return attr.evolve(self, partial_kwargs=kwargs) 
```

While _validate_arg_names(), only the parameter for task group function is allow for kwargs.
For example, my test code is:
```python
@task_group
def tg(who: str):
    tgt1 = BashOperator(task_id="dummy1", bash_command="sleep 300")
    tgt2 = BashOperator(task_id="dummy2", bash_command="echo done")
    tgt1 >> tgt2
```
So only parameter 'who' is allowed for kwargs.
Seems the partial() is designed to pass parameters to task group function, not for expand execution... 

So I think expand() may be still the best place to add concurrency limit parameter.

How do you think about this?

Reply:
Don’t add it to `kwargs`, do it in a separate argument like you do in `expand`. That is the right way to do it. Source: I implemented both `partial` and `expand`.

Reply:
btw, airflow code is not compatible with pydantic 2.x, which blocked me a long time for debugging in local machine, T___T

Reply:
Thinking this again, do this in _neither_ `partial` and `expand` makes sense, since those target the parameters of the _decorated function_, not _task_ parameters. The best way to do it is likely in the `@task_group` decorator—yes this makes the argument appear when the task is not mapped, but in the case it can simply be a no-op, which should be fine as long as it is appropriately documented. This would also make it possible to use the argument in non-decorator version of the task group.

Reply:
Good idea!
This is consistent with the usage of max_active_tis_per_dagrun for task. 

So is it better to use the parameter name as `max_active_tgs_per_dagrun`?
tgs may be not so good, but I did not find any work better...

Reply:
Maybe `groups`? It’s pretty obviously task groups, after all?

Reply:
Either just put it in `@task_group(...)`, or use `.override()`

Reply:
got it, thank you!